src/test/java/org/gridsuite/modification/server/LinesAttachToSplitLinesTest.java

```diff
@@ -0,0 +1,255 @@
+    private Network network;
+
+    @Before
+    public void setUp() {
+        objectWriter = mapper.writer().withDefaultPrettyPrinter();
+        // /!\ create a new network for each invocation (answer)
+        when(networkStoreService.getNetwork(TEST_NETWORK_ID)).then((Answer<Network>) invocation -> {
+            network = NetworkCreation.create(TEST_NETWORK_ID, true);
+            return network;
+        });
+        networkModificationService.setReportServerRest(reportServerRest);
+        given(reportServerRest.exchange(eq("/v1/reports/" + TEST_REPORT_ID), eq(HttpMethod.PUT), ArgumentMatchers.any(HttpEntity.class), eq(ReporterModel.class)))
+                .willReturn(new ResponseEntity<>(HttpStatus.OK));
+
+        // clean DB
+        modificationRepository.deleteAll();
+    }
+
+    @After
+    public void tearOff() {
+        // clean DB
+        modificationRepository.deleteAll();
+    }
+
+    private List<LinesAttachToSplitLinesInfos> getModifications() {
+        return modificationRepository.getModifications(TEST_GROUP_ID, false, true)
+                .stream().map(LinesAttachToSplitLinesInfos.class::cast).collect(Collectors.toList());
+    }
+
+    private void createModification() throws Exception {
+        LinesAttachToSplitLinesInfos linesAttachToSplitLines = LinesAttachToSplitLinesInfos.builder()
+                .type(ModificationType.LINES_ATTACH_TO_SPLIT_LINES)
+                .lineToAttachTo1Id("line1")
+                .lineToAttachTo2Id("line2")
+                .attachedLineId("line3")
+                .voltageLevelId("v4")
+                .bbsBusId("1.A")
+                .replacingLine1Id("nl4")
+                .replacingLine1Name("NewLine4")
+                .replacingLine2Id("nl5")
+                .replacingLine2Name("NewLine5")
+                .build();
+        String linesAttachToSplitLinesJson = objectWriter.writeValueAsString(linesAttachToSplitLines);
+        mockMvc.perform(post(CREATE_URI_STRING, TEST_NETWORK_ID).content(linesAttachToSplitLinesJson).contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk()).andReturn();
+    }
+
+    @Test
+    public void testCreate() throws Exception {
+        createModification();
+
+        List<LinesAttachToSplitLinesInfos> modifications = getModifications();
+        assertEquals(1, modifications.size());
+        LinesAttachToSplitLinesInfos modification = modifications.get(0);
+        assertEquals("line1", modification.getLineToAttachTo1Id());
+        assertEquals("line2", modification.getLineToAttachTo2Id());
+        assertEquals("line3", modification.getAttachedLineId());
+        assertEquals("v4", modification.getVoltageLevelId());
+        assertEquals("NewLine4", modification.getReplacingLine1Name());
+        assertEquals("NewLine5", modification.getReplacingLine2Name());
+    }
+
+    @Test
+    public void testUpdate() throws Exception {
+        // 1- First create a modification
+        createModification();
```

Comment:
I think this method should only test the update and not the creation, and you should insert the network modification directly via the repository instead of via the API.

Reply:
yes, why not. at the end we have to create a modification.
I will change it